module-2/backend/week_two.md

```diff
@@ -26,20 +39,82 @@ end
 
+@team.owners.where(owner_id: 4)
+
 5. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+Teacher has_many students
+Student belongs_to a Teacher
+
+Teacher              Student
+id----------         id
+name       |         name 
+department _________ teacher_id
+
 6. Define foreign key, primary key, and schema.
+
+- Primary Key is a unique identifier for a given table.
+- Foreign Key is an attribute that sits on a second table to create the relationship to the first table's primary key.
+- A schema is a layout of the attributes(columns) associated to a given table and their values(string,text,integer,date). Can be found in your database folder of your project directory.
+
 7. Describe the relationship between a foreign key on one table and a primary key on another table.
+
+I believe this answer is similiar to the previous one-> a foriegn key sits on a second table and is the unique id of the first table. That's how the relationship is created.
+
 8. What are the parts of an HTTP response?
 
+- A status line 
+- A status code (100-informational, 200-success, 300-redirection, 400-client error, 500-server error)
+- Header
 
 ### Optional Questions
 
 1. Name your five favorite ActiveRecord methods (i.e. methods your models inherit from ActiveRecord) and describe what they do.
+.has_attribute?(attribute_name) = returns true if the attribute exists, false if it does not
+.to_param = generates a clean url with the id and attribute name
+.not(argument) = selects from table using WHERE + NOT
+.reflect_on_all_associations = returns an array of all associations. You can specify if you want to reflect on :has_many,                                    :belongs_to, etc.
+.has_secure_token(token) = generates a unique 24 character token
+
 2. Name your three favorite ActiveRecord rake tasks and describe what they do.
-3. What two columns does `t.timestamps null: false` create in our database?
+
+rake notes = will search through code for comments beginning with FIXME, OPTIMIZE or TODO.
+rake routes = shows all of the defined routes
+rake doc:rails = will generate API docs in your application 
+
+3. What two columns does `t.timestamps null: false` create in our database? 
+
+created_at
+updated_at
+
 4. In a database that's holding schools and teachers, what will be the relationship between schools and teachers?
+A School has_many teachers
+A Teacher belongs_to a school
+
 5. In the same database, what will you need to do to create this relationship (draw a schema diagram)?
+A primary key and foreign key
+
+School         Teacher
+id -----       id
+name   |       name
+city   |       grade
+state  |______ school_id
+
+
 6. Give an example of when you might want to store information besides ids on a join table.
+
+To be honest, I don't have one. I thought the purpose of a join table was to normalize and keep the schema and db well designed.
```

Comment:
`invoice_items` is a good example: holds quantity and unit price in addition to the foreign keys. Could also store a grade for a class on a join between students and teachers.